plugins/ingest/src/main/java/org/elasticsearch/ingest/TemplateService.java

```diff
@@ -0,0 +1,70 @@
+package org.elasticsearch.ingest;
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Map;
+
+public interface TemplateService {
+
+    default Template compile(String template) {
+        if (template != null) {
```

Comment:
I think right now the only thing that we assume is that the template markers are `{{` and `}}`. Whether mustache or another templating engine is used is up to the runtime the pipeline framework runs in.

So with that in mind I do think it is ok to have logic that decides whether we delegate to the actual template service implementation. But maybe we should do this in a different place then here?